### ISSUE TITLE
How to understand the code about output feature？

### ISSUE THREAD

**Issue by FlyuZ** [layumi/Person_reID_baseline_pytorch#310]
I have some doubts while reading the code.  hope you can answer
In train.py  line 270
`
fnorm = torch.norm(ff, p=2, dim=1, keepdim=True)
ff = ff.div(fnorm.expand_as(ff))
`
In test.py
`
features = torch.cat((features,ff.data.cpu()), 0)
`
What is the meaning of this code?  Which one can represent the feature of the input image？
Why not put the code in ClassBlock?

Thanks！

**Comment by layumi** [comment]
@FlyuZ
Thank you. 
 
Actually, we also norm the feature in the test.py 

The `features` store the all features of the gallery images or all query images. We need to concatenate the feature to a matrix of `N * D` (D is 512 or 2048, N is the number of total input samples) for the final ranking process.

**Comment by FlyuZ** [comment]
Thanks for your answer, and I would like to ask, why is the output feature divided by the 2-norm? Are there any related paper recommendations?
